src/transport/net_ib.cc

```diff
@@ -1232,28 +1544,41 @@ ncclResult_t ncclIbIrecv(void* recvComm, int n, void** data, int* sizes, int* ta
   NCCLCHECK(ncclIbGetRequest(&comm->verbs, &req));
   req->type = NCCL_NET_IB_REQ_RECV;
   req->sock = &comm->sock;
+  req->addr = comm->addr;
   req->nreqs = n;
+  
   if (comm->gidInfo.link_layer == IBV_LINK_LAYER_ETHERNET) req->gidInfo = &comm->gidInfo;
   for (int i=0; i<n; i++) req->recv.sizes[i] = 0;
-
-  struct ibv_recv_wr wr;
-  memset(&wr, 0, sizeof(wr));
-  wr.wr_id = req - comm->verbs.reqs;
-
-  wr.sg_list = NULL;
-  wr.num_sge = 0;
-
-  TIME_START(1);
   const int nqps = ncclParamIbSplitDataOnQps() ? comm->nqps : 1;
-  for (int q=0; q<nqps; q++) {
-    struct ibv_qp* qp = comm->qps[comm->qpIndex];
-    struct ibv_recv_wr* bad_wr;
-    NCCLCHECK(wrap_ibv_post_recv(qp, &wr, &bad_wr));
-    comm->qpIndex = (comm->qpIndex+1)%comm->nqps;
+  if (!getIbDisableStatus(comm->addr)) 
+  {
+    struct ibv_recv_wr wr;
+    memset(&wr, 0, sizeof(wr));
+    wr.wr_id = req - comm->verbs.reqs;
+
+    wr.sg_list = NULL;
+    wr.num_sge = 0;
+
+    TIME_START(1);
+    for (int q=0; q<nqps; q++) {
+      struct ibv_qp* qp = comm->qps[comm->qpIndex];
+      struct ibv_recv_wr* bad_wr;
+
+      ncclResult_t nRet = wrap_ibv_post_recv(qp, &wr, &bad_wr);
+      if (nRet != ncclSuccess)
+      {
+        // in resilient mode, if post receive failed, just disable the ib peer, and will contiue to post the fifo via socket using timeout logic to handle receive
+        if (!ncclParamResilientEnabled())  
+        {
+          return nRet;
+        }
+        disableIb(comm->addr);
```

Comment:
> disableIb(comm->addr); [](http://example.com/codeflow?start=7&length=23)

Question here: looks like we will process all of the QPs here and may disable the comm->addr multiple times, why not disable the addr when there is error happening when calling post_recv? Besides, when we will also call ncclIbPostFifo even calling post_recv fails and device is disabled, is this logic as same as the original logic?